accepted/2020/workloads/workload-manifest.md

```diff
@@ -0,0 +1,430 @@
+
+For example, an SDK pack might contain a compiler that's a native executable and hence depends on the host platform:
+
+```json
+"foo.sdk.compiler": {
+    "version": "1.0",
+    "kind": "sdk",
+    "alias-to": {
+        "osx-x64": "foo.sdk.compiler.mac",
+        "win-x64": "foo.sdk.compiler.windows"
+    }
+}
+```
+
+Workload definitions and SDK imports would refer to this as `"foo.sdk.compiler"`, but on Windows the pack that gets installed and resolved would be `"foo.sdk.compiler.windows"`. The `"foo.sdk.compiler.windows"` pack would be present on disk and its ID would be visible in resolved paths, but otherwise transparent and not able to be used directly.
+
+Another use of aliases is to install multiple version bands of the same pack by synthesizing new versioned pack IDs:
+
+```json
+"foo.framework": {
+    "version": "2.0.4",
+    "kind": "framework",
+    "alias-to": {
+        "*": "foo.framework",
+    }
+},
+"foo.framework.1": {
+    "version": "1.3.2",
+    "kind": "framework",
+    "alias-to": {
+        "*": "foo.framework",
+    }
+},
+```
+
+A workload or framework reference that referenced `"foo.framework"` would get `"foo.framework"` version `2.0.4`, while a workload or framework reference that referenced `"foo.framework.1"` would get `"foo.framework"` version `1.3.2`,
+
+## Example Platform SDK Manifest
+
+Here is a *hypothetical* example manifest. It's not prescriptive but demonstrates concepts an patterns that can be used by platform implementations.
+
+```json5
+{
+    "version": 5,
+    "workloads": {
+        // this is a dev workload that would typically be installed by a developer getting
+        // started with this platform. it's composed of several smaller build workloads
+        // plus a template pack for creating projects
+        "xamarin-android": {
+            "description": "Create, build and run Android apps",
+            "kind": "dev",
+            "packs": [
+                "Xamarin.Android.Templates",
+            ],
+            // on dev machines we expect to pre-install support
+            // for common device architectures
+            "extends": [
+                "xamarin-android-build",
+                "xamarin-android-build-armv7a",
+                "xamarin-android-build-x86"
+            ],
+        },
+        "xamarin-android-build": {
+            "description": "Build and run Android apps",
+            "packs": [
+                "Xamarin.Android.Sdk",
+                "Xamarin.Android.BuildTools",
+                "Xamarin.Android.Framework",
+                "Xamarin.Android.Runtime",
+                "Mono.Android.Sdk"
+            ]
+        },
+        // on CI machines, this will only be installed if
+        // the app actually targets the armv7a architecture
+        "xamarin-android-build-armv7a": {
```

Comment:
There is clearly a lot of convention at play. That's good, however, it would be good to see that more defined.

Reply:
Yeah. This is maybe getting out of the scope of this doc though as it's more about how to structure the packs that make up a workload. Maybe something for an authoring guide?